src/utils/assets.ts

```diff
@@ -16,9 +16,29 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import TokensMetadata from '@alephium/token-list'
+import { calcTxAmountsDeltaForAddress } from '@alephium/sdk'
+import { Transaction } from '@alephium/sdk/api/explorer'
+import TokensMetadata, { ALPH } from '@alephium/token-list'
 
 import { NetworkType } from '@/types/network'
 
+import { convertToPositive } from './numbers'
+
 export const getAssetInfo = ({ assetId, networkType }: { assetId: string; networkType: NetworkType }) =>
   TokensMetadata[networkType].tokens.find((tm) => tm.id === assetId)
+
+export const getAddressAssetsWithAmounts = ({
+  transaction,
+  addressHash,
+  networkType
+}: {
+  transaction: Transaction
+  addressHash: string
+  networkType: NetworkType
+}) => {
+  const { alph: alphAmount, tokens: tokenAmounts } = calcTxAmountsDeltaForAddress(transaction, addressHash)
+  const amount = convertToPositive(alphAmount)
+  const tokens = tokenAmounts.map((token) => ({ ...token, amount: convertToPositive(token.amount) }))
+  const tokenAssets = [...tokens.map((token) => ({ ...token, ...getAssetInfo({ assetId: token.id, networkType }) }))]
```

Comment:
nit: instead of generating 3 arrays, this can be achieved with only 1 (but this is VERY NIT NIT NIT 😅)

```suggestion
const tokenAssets = tokenAmounts.map((token) => ({
    ...token,
    ...getAssetInfo({ assetId: token.id, networkType }),
    amount: convertToPositive(token.amount),
  }))
```

Reply:
I like it!